lcov_parser.rb

```diff
@@ -45,11 +45,14 @@ def self.parse(filepath)
                         executableLines: 0,
                         lineCoverage: 0,
                         functions: [] }
-        target[:name] = Regexp.last_match(1).strip.empty? ? "No Name" : Regexp.last_match(1).strip
+        tn_value = Regexp.last_match(1).strip
+        target[:name] = tn_value.empty? ? "No Name" : tn_value
 
       when /^SF:(.+)/ # Source file
         current_filename = $LAST_MATCH_INFO[1].gsub(%r{^\./}, "")
+        file_name_only = File.basename(current_filename)
         target_file[:name] = current_filename
+        target[:name] = file_name_only if target[:name] == "No Name"
```

Comment:
This variable is not used anywhere else so you can also use it as follows (It's up to you):
```suggestion
        target[:name] = File.basename(current_filename) if target[:name] == "No Name"
```

 But if you do not want to change, can you use these lines consecutively.
Like that:
```ruby
        current_filename = $LAST_MATCH_INFO[1].gsub(%r{^\./}, "")
        target_file[:name] = current_filename
        file_name_only = File.basename(current_filename)
        target[:name] = file_name_only if target[:name] == "No Name"
```